packages/react-components/react-carousel-preview/library/src/components/CarouselAutoplayButton/useCarouselAutoplayButton.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { slot, useControllableState, useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import type { CarouselAutoplayButtonProps, CarouselAutoplayButtonState } from './CarouselAutoplayButton.types';
+import { useButton_unstable } from '@fluentui/react-button';
+import { PlayCircleRegular, PauseCircleRegular } from '@fluentui/react-icons';
+import { useEventCallback } from '@fluentui/react-utilities';
+import { mergeCallbacks } from '@fluentui/react-utilities';
+import { useCarouselContext_unstable as useCarouselContext } from '../CarouselContext';
+import { ARIAButtonElement } from '@fluentui/react-aria';
+
+/**
+ * Create the state required to render CarouselAutoplayButton.
+ *
+ * The returned state can be modified with hooks such as useCarouselAutoplayButtonStyles_unstable,
+ * before being passed to renderCarouselAutoplayButton_unstable.
+ *
+ * @param props - props from this instance of CarouselAutoplayButton
+ * @param ref - reference to root HTMLDivElement of CarouselAutoplayButton
+ */
+export const useCarouselAutoplayButton_unstable = (
+  props: CarouselAutoplayButtonProps,
+  ref: React.Ref<ARIAButtonElement>,
+): CarouselAutoplayButtonState => {
+  const [autoplay, setAutoplay] = useControllableState({
```

Comment:
Usually, this would go in the useCarousel hook with default/controlled values passed in to the Carousel wrapper - however autoplay has a regulations requirement that it can ONLY be enabled if a button to stop it is also present.

By keeping the state local to this component, we can ensure this regulation is followed.